src/GitVersionCore/GitVersionContext.cs

```diff
@@ -10,7 +10,9 @@
     public class GitVersionContext
     {
         public GitVersionContext(IRepository repository, string targetBranch, Config configuration, bool onlyEvaluateTrackedBranches = true, string commitId = null)
-             : this(repository, repository.Branches.SingleOrDefault(b => b.CanonicalName == targetBranch || b.FriendlyName == targetBranch) ?? repository.Head, configuration, onlyEvaluateTrackedBranches, commitId)
+             : this(repository,
+                   repository.Branches.SingleOrDefault(b => b.CanonicalName == targetBranch || b.FriendlyName == targetBranch) ?? repository.Head,
+                   configuration, onlyEvaluateTrackedBranches, commitId)
```

Comment:
I only now discovered that this all happens within a constructor. Sorry for not catching that earlier. Could you please add a static method that wraps `repository.Branches.SingleOrDefault(b => b.CanonicalName == targetBranch || b.FriendlyName == targetBranch) ?? repository.Head` and invoke that method from the constructor?